### DISCUSSION TITLE
nested tables do a lot of reading.

### DISCUSSION THREAD

**Discussion by solderland** [prisma/prisma#12023] [Q&A]
Hello, i have 4 relational tables.
I am using mysql and hosting on planetscale.

The result I got with the following code:

```ts
const mines = await Mine.findMany({
    where: {
        deleted: false,
    },
    // include: {
    //   users: {
    //     include: {
    //       bunny: {
    //         select: {
    //           owner: true,
    //         },
    //       },
    //     },
    //   },
    // },
});
```

 -> Reads: 8.6 rows/s
 
 when you activate the relational tables:
 ```ts
const mines = await Mine.findMany({
    where: {
        deleted: false,
    },
    include: {
        users: {
            include: {
                bunny: {
                    select: {
                        owner: true,
                    },
                },
            },
        },
    },
});
 ```
 -> Reads: 1.6k rows/s
 
 I don't know how to optimize this, please suggest a solution.
 
 My Models: 
 ```prisma
 model Mine {
  id                 String           @id @unique(sort: Desc) @default(cuid())
  name               String
  image              String
  imageType          String
  x                  Float
  y                  Float
  width              Float?
  height             Float?
  frame              Int?
  speed              Int?
  type               String
  startTime          DateTime         @default(now())
  assignEndTime      Int
  processTime        Int
  rewardProcessTime  Int
  users              MineWithBunny[]
  job                Job[]
  isDistributeReward Boolean          @default(false)
  rewards            MineWithReward[]
  distributedRewards UserWithReward[]
  deleted            Boolean          @default(false)
  createdAt          DateTime         @default(now())
  updatedAt          DateTime         @default(now()) @updatedAt
}
 ```
 
 ```prisma
 model User {
  id                 String               @id @unique(sort: Desc) @default(cuid())
  discordId          String               @unique(sort: Desc)
  username           String
  avatar             String?
  discordAccessToken String
  walletAddress      String?              @unique(sort: Desc)
  isAdmin            Boolean?             @default(false)
  isHolder           Boolean?             @default(false)
  bunnies            Bunny[]
  rewards            UserWithReward[]
  withdrawSignatures WithDrawSignatures[]
  createdAt          DateTime             @default(now())
  updatedAt          DateTime             @default(now()) @updatedAt
}
 ```
 
 ```prisma
 model Bunny {
  id         String          @id @unique(sort: Desc) @default(cuid())
  name       String
  address    String          @unique(sort: Desc)
  image      String
  power      Int
  user       User?           @relation(fields: [owner], references: [walletAddress])
  owner      String?
  mines      MineWithBunny[]
  assignable Boolean         @default(false)
  createdAt  DateTime        @default(now())
  updatedAt  DateTime        @default(now()) @updatedAt
  stakedAt   DateTime        @default(now())


  @@index([owner(sort: Desc)])
}
 ```
 
 Thanks !

**Comment by austincrim** [comment]
Hey @solderland 👋 , thanks for the interesting question.

Could you enable [query logging](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/logging#log-to-stdout) to see the generated SQL for each query?

```ts
const prisma = new PrismaClient({ log: ['query'] })
```

How are you counting the rows read?

Thanks!

**Comment by solderland** [comment]
Hi @austincrim 
Planetscale is counting.
<img width="463" alt="Screen Shot 2022-02-25 at 09 53 27" src="https://user-images.githubusercontent.com/94372274/155668889-a9c1b402-04cb-442e-8562-24f0898814a0.png">

query:
<img width="1243" alt="Screen Shot 2022-02-25 at 10 04 45" src="https://user-images.githubusercontent.com/94372274/155670186-cdf3daf7-d93e-4c4d-a873-74996bcf1fa7.png">

thanks sir!

**Reply by pantharshit00** [reply]
Hello @solderland, our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

This is related to https://github.com/prisma/prisma/issues/10993. Currently, there no way of configuring the generated queries. This will likely be optimised in the future.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
